vllm/v1/engine/core.py

```diff
@@ -141,6 +157,21 @@ def profile(self, is_start: bool = True):
     def reset_prefix_cache(self):
         self.scheduler.reset_prefix_cache()
 
+    def calculate_grammar_bitmasks(self):
+        for req in self.scheduler.running:
+            # ignore requests that doesn't use guided decoding
+            # or ignore requests that grammar is not ready
+            if not req.use_guided_decoding or not req.is_grammar_ready:
+                continue
+
+            # Check if grammar is ready in cache
+            grammar = self.guided_decoding_manager.get(req)
+            if grammar is not None:
+                req.grammar = grammar
+                req.allocate_bitmask(1,
+                                     self.guided_decoding_manager.vocab_size)
+                continue
```

Comment:
There is no need to go through all of the scheduler request to see which ones are using guided decoding. The `guided_decoding_manager` should know all of the requests from `self.guided_decoding_manager.should_cache(req)` above or within the Scheduler's `guided_decoding_request_ids` list. I think we can get the requests of interest from the manager and just check their RequestStatus